File: core-examples/src/main/java/io/vertx/example/core/verticle/worker/WorkerVerticle.java
package io.vertx.example.core.verticle.worker;

import io.vertx.core.AbstractVerticle;

/**
 * An example of worker verticle
 */
public class WorkerVerticle extends AbstractVerticle {


  @Override
  public void start() throws Exception {
    System.out.println("[Worker] Starting in " + Thread.currentThread().getName());

    vertx.eventBus().<String>consumer("sample.data", message -> {
      System.out.println("[Worker] Consuming data in " + Thread.currentThread().getName());
      String body = message.body();
      message.reply(body.toUpperCase());
    });
  }
}

File: core-examples/src/main/kotlin/io/vertx/example/core/verticle/worker/WorkerVerticle.kt
package io.vertx.example.core.verticle.worker


class WorkerVerticle : io.vertx.core.AbstractVerticle()  {
  override fun start() {
    println("[Worker] Starting in ${java.lang.Thread.currentThread().getName()}")

    vertx.eventBus().consumer<String>("sample.data", { message ->
      println("[Worker] Consuming data in ${java.lang.Thread.currentThread().getName()}")
      var body = message.body()
      message.reply(body.toUpperCase())
    })
  }
}
